locals.tf

```diff
@@ -0,0 +1,37 @@
+# Local Values
+# https://www.terraform.io/docs/language/values/locals.html
+
+locals {
+
+  # The variable for the workspace is only for testing purposes, it should not be set under any other use case.
+
+  workspace = var.workspace != null ? var.workspace : terraform.workspace
+
+  # The regex is used to parse the workspace name into its components, the components are used to set the region, zone, and environment variables.
+
+  #workspace_regex = "^(?:(?P<region>us-[a-z]+\\d+)(?:-(?P<zone>[abcd]))?(?:-[a-zA-Z0-9-]+)?|main)-(?P<environment>sandbox|non-production|production)$"
+  workspace_regex = "^(?:(?P<region>us-[a-z]+\\d+)(?:-(?P<zone>[abcd]))?(?:-[a-zA-Z0-9-]+)?|main)-(?P<environment>sandbox|non-production|production)$"
```

Comment:
@coderabbitai This code works for all of these workspace names: main-sandbox, us-east1-a-sandbox, us-east1-a-test-sandbox, main-non-production but doesn't work for this: us-east1-a-non-production. In the failing use case, I return production as an environment instead of non-production. I suspect it's something with the hyphen in non-production. Any ideas?

Reply:
No, that's not it; changing the placement doesn't help.

Reply:
I have the same problem with the most recent commit. Can you check?

Reply:
That didn't work; this made all the other working tests fail.

Reply:
We had to separate the regex to make this work; it's also a bit easy to read. Look at it again and let me know if it can be improved.